server/src/stateManagement.mjs

```diff
@@ -201,9 +202,27 @@ function getState(userId) {
       }
     }
     if (userId in stateCopy){
+      let appId, user
       const userState = stateCopy[''+userId];
-      resetSequenceStateValues(finalState, userState);
-      mergeWith(finalState, userState, mergeCustomizer);
+      user = userId.split("~")[0];
```

Comment:
There is a function called "parseUser" in userManagement that will automatically split the userId into an object containing user, group, and appId. Please use this to avoid duplicating code